### PR TITLE
DAOS-16469 dtx: properly handle DTX partial commit

### PR DESCRIPTION
**Pull request by Nasf-Fan**
When a DTX leader globally commit the DTX, it is possible that some DTX participant(s) cannot commit such DTX entry because of kinds of issues, such as network or space trouble. Under such case, the DTX leader needs to keep the active DTX entry persistently for further commit/resync. But it does not means related modification attched to such DTX entry on the leader target cannot be committed, instead, we can commit related modification with only keeping the DTX header. That is enough for the DTX leader to do further DTX commit/resync to handle related former failed DTX participant(s).

The benefit is that VOS aggregation on the leader target will not be affected by remote DTX commit failure.

Allow-unstable-test: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/object/srv_obj.c
Comment (liuxuezhao) [inline src/object/srv_obj.c:2030]: if retry > 30 times for fetch, -DER_INPROGRESS will return to client?
then client will retry again right? do you think it is better than always call refresh here?

**Review by liuxuezhao — DISMISSED** [review]

File: src/object/srv_obj.c
Comment (Nasf-Fan) [reply src/object/srv_obj.c:2030]: Yes, if `dtx_refresh` failed so frequently, we will stop the server-side retry and return -DER_INPROGRESS to client. Not sure whether it is better, but It will at least reduce the possibility of too much ULTs caused system busy.

**Comment by Nasf-Fan** [comment]
Ping reviewers, thanks!

**Review by wangshilong — APPROVED** [review]

**Review by liuxuezhao — APPROVED** [review]

**Comment by daltonbohning** [comment]
> Ping @daos-stack/daos-gatekeeper , thanks!

Ideally should have an approval from @daos-stack/vos-owners before merging, or I will defer to @gnailzenh on DTX PRs since I am not familiar enough.

**Comment by Nasf-Fan** [comment]
Ping @gnailzenh

**Pull request merged**
